.github/workflows/builds.yml

```diff
@@ -28,4 +28,8 @@ jobs:
       - name: Configure
         run: cmake -DCMAKE_BUILD_TYPE=${{ matrix.build }} -B build -S .
       - name: Build
-        run: cmake --build build --
+        run: >
+          cmake --build build --
+          && cmake --build build -- test
+          && ./run_examples.sh build
```

Comment:
At the risk of sounding pedantic, could we split this into three different jobs? Keep the build job as it is, then run a `Tests` job to execute just `cmake --build build -- test`, and then an `Examples` job that runs `./run_examples.sh build`?

Reply:
@stephenswat because then you start having to deal with artifacts to ship the build output between jobs, which in my opinion is not worth it.

@krasznaa if you care about the output, I would probably prefer setting `CTEST_OUTPUT_ON_FAILURE=1` on the job, like we do in ACTS core.

Reply:
@stephenswat what I can do though is split it into steps, that way the interfaces shows them separately.

Reply:
Sorry - I didn't mean different jobs but rather different steps of the same job.

Reply:
@stephenswat ok done.

Reply:
I'm okay with using `CTEST_OUTPUT_ON_FAILURE`. But since the test is failing at the moment, that quite strongly underlines my point of output being necessary in the CI on failure...